src/DataStructures/Variables.hpp

```diff
@@ -196,6 +201,36 @@ class Variables<tmpl::list<Tags...>> {
   void initialize(size_t number_of_grid_points, value_type value);
   /// @}
 
+  /// @{
+  /// Set the VectorImpl to be a reference to another VectorImpl object
+  void set_data_ref(const gsl::not_null<Variables*> rhs) {
+    set_data_ref(rhs->data(), rhs->size());
+  }
+
+  void set_data_ref(pointer const start, const size_t size) {
```

Comment:
Not necessarily wrong, but this just bit me so we should discuss it.  The non-owning constructor and `set_data_ref` take the same argument types, but they mean different things, which I find surprising.  In the constructor, the second argument is the number of grid points, but in `set_data_ref` it is the storage size, i.e., the number of grid points times the number of independent components.

Reply:
oh, yea that should be consistent... I don't have a preference. I think the argument for `number_of_grid_points` is this is consistent with the value constructors, while the argument for consistently using `allocation_size` is it doesn't assume the buffer is of the correct size. Of course the latter would likely lead to segfaults or ASAN issues. Do you have any preference @wthrowe or @kidder?

Reply:
They both make sense.  Let's go with allocation size, since (pointer, length) is a really common general interface.

Reply:
Okay, sounds good to me! I agree with that :)